src/features/tokens/routes/types.ts

```diff
@@ -3,22 +3,52 @@ export enum RouteType {
   CollateralToSynthetic = 'collateralToSynthetic',
   SyntheticToSynthetic = 'syntheticToSynthetic',
   SyntheticToCollateral = 'syntheticToCollateral',
+  IbcNativeToIbcNative = 'ibcNativeToIbcNative',
+  IbcNativeToHypNative = 'ibcNativeToHypNative',
 }
 
-export interface Route {
+interface BaseRoute {
   type: RouteType;
   // The underlying 'collateralized' token:
   baseTokenCaip19Id: TokenCaip19Id;
-  baseRouterAddress: Address;
   originCaip2Id: ChainCaip2Id;
-  originRouterAddress: Address;
   originDecimals: number;
   destCaip2Id: ChainCaip2Id;
-  destRouterAddress: Address;
   destDecimals: number;
   // The underlying token on the destination chain
-  // Only set for CollateralToCollateral routes (b.c. sealevel need it)
+  // Only set for CollateralToCollateral routes (b.c. sealevel needs it)
   destTokenCaip19Id?: TokenCaip19Id;
 }
 
+export interface HypRoute extends BaseRoute {
+  type:
+    | RouteType.CollateralToCollateral
+    | RouteType.CollateralToSynthetic
+    | RouteType.SyntheticToCollateral
+    | RouteType.SyntheticToSynthetic;
+  baseRouterAddress: Address;
+  originRouterAddress: Address;
+  destRouterAddress: Address;
+}
+
+interface BaseIbcRoute extends BaseRoute {
+  originIbcDenom: string;
+  sourcePort: string;
+  sourceChannel: string;
+  derivedIbcDenom: string;
+}
+
+export interface IbcRoute extends BaseIbcRoute {
+  type: RouteType.IbcNativeToIbcNative;
+}
+
+export interface IbcToHypRoute extends BaseIbcRoute {
```

Comment:
Nit: Wonder if `IbcToWarpRoute` and `RouteType.IbcNativeToWarp` would be a bit more clear